torch/csrc/jit/codegen/cuda/graph_fuser.cpp

```diff
@@ -1365,10 +1371,59 @@ void guardFusionGroup(Node* fusion) {
 
   versioning_if->insertAfter(typecheck_node);
 
+  auto fusion_graph = fusion->g(attr::Subgraph);
+  std::vector<Value*> check_flags = {};
+
+  std::cout << "profiled ivalue\t" << profiled_ivalue_indices.size() << std::endl;
+  for (auto node : fusion_graph->nodes()) {
+    if (node->kind() == prim::view_copy) {
+      auto self_type = node->inputs().front()->type()->cast<TensorType>();
+      auto self_sizes = getTensorSizes(self_type);
+
+      std::cout << fusion->owningGraph()->toString() << std::endl;
+      std::cout << node->inputs().back()->owningGraph()->toString() << std::endl;
+
+      // TODO: refactor - map subgraph input to fusion input
+      // TODO: refactor - remove corresponding prim::profile_ivalue node
+      auto shape_node = node->inputs().back()->node();
+      auto param_node = shape_node->inputs().front()->node();
+      int index = 0;
+      for (auto i : c10::irange(fusion_graph->inputs().size())) {
+        if (fusion_graph->inputs()[i]->node() == param_node) {
+          index = i;
+        }
+      }
+      auto profile_ivalue = fusion->inputs()[index];
+
+      auto view_sizes = constant_as<c10::List<int64_t>>(node->inputs().back());
+      TORCH_INTERNAL_ASSERT(view_sizes.has_value());
+
+      // Get size constraint given self_sizes and view_sizes
+      auto constraint = analyzeViewConstraint(self_sizes, view_sizes->vec());
+      auto constrain_const = fusion->owningGraph()->insertConstant(IValue(constraint));
+      constrain_const->node()->moveBefore(versioning_if);
+
+      // TODO: Get input tensors that correspond with self tensor
+      auto input = tensor_inputs_to_check.front();
```

Comment:
This one is pretty tricky.

I think you are only interested in the shape of `self` tensor, we have something built in an earlier stage `buildShapeExpressions`, we might need some extra plumbing for that to get it here.

Reply:
That sounds perfectly fine. Let's do the refactor in separate PRs as well.